packages/cozy-client/src/helpers.js

```diff
@@ -18,3 +19,46 @@ export const dehydrate = document => {
   )
   return dehydrated
 }
+
+const ensureFirstSlash = path => {
+  if (!path) {
+    return '/'
+  } else {
+    return path.startsWith('/') ? path : '/' + path
+  }
+}
+
+/**
+ * generateWebLink - Construct a link to a web app
+ *
+ * @param {object} Options               Object of options
+ * @param {string}   options.cozyUrl       Base URL of the cozy, eg. cozy.tools or test.mycozy.cloud
+ * @param {Array}    options.searchParams  Array of search parameters as [key, value] arrays, eg. ['username', 'bob']
+ * @param {string}   options.nativePath    Path inside the app, eg. /files/test.jpg
+ * @param {string}   options.slug          Slug of the app
+ * @param {string}   options.subDomainType Whether the cozy is using flat or nested subdomains. Defaults to flat.
+ *
+ * @returns {string} Generated URL
+ */
+export const generateWebLink = ({
+  cozyUrl,
+  searchParams = [],
+  nativePath,
+  slug,
+  subDomainType
+}) => {
+  nativePath = ensureFirstSlash(nativePath)
+  const url = new URL(cozyUrl)
+  url.host =
+    subDomainType === 'nested'
+      ? `${slug}.${url.host}`
+      : url.host
+          .split('.')
+          .map((x, i) => (i === 0 ? x + '-' + slug : x))
+          .join('.')
+  for (const [param, value] of searchParams) {
+    url.searchParams.set(param, value)
+  }
+  url.hash = nativePath
+  return url.toString()
```

Comment:
Nit: It feels like the ergonomic of this function are a bit off. The subDomainType could be taken from the client, the cozyURL too, this would simplify the options.

Reply:
Yep I plan to rewrite this method specially since this method is now living in CozyClient. 

We still need to mimic https://github.com/cozy/cozy-client/blob/61f5de62a98ac58d04a09d2d955842c4fdfec638/packages/cozy-client/src/CozyClient.js#L1176 for an OAuth client ;)

Reply:
This method will be used to build urls that point to another Cozy than the one used by the client.
This is why I put this function in the `helpers` module.